### PR TITLE
Add more modes to infobox

### PR DESCRIPTION
**Pull request by pickfire**
I think only view mode left.

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -3443,62 +3425,12 @@ fn view_mode(cx: &mut Context) {
...
-    })
+fn select_textobject_around(cx: &mut Context) {
+    select_textobject(cx, textobject::TextObject::Around);
 }
 
-use helix_core::surround;
-use helix_core::textobject;
-
-fn match_mode(cx: &mut Context) {
-    let count = cx.count;
```
Comment (pickfire) [inline helix-term/src/commands.rs:3482]: I tried adding this but doesn't seemed to make a difference for me.

Comment (sudormrfbin) [reply helix-term/src/commands.rs:3482]: `on_next_key` always passes the count as None, so it's necessary for counts to work properly as of now:

https://github.com/helix-editor/helix/blob/a091e113bca9cef99f11373de04520ead906f3c0/helix-term/src/ui/editor.rs#L653-L664

Using the previous count when constructing `Context` if there's a registered `on_next_key` callback is the definitive fix anyway, which can be done later I suppose (count will be broken till then though).

Comment (pickfire) [reply helix-term/src/commands.rs:3482]: I did this but noticed nothing different.

Comment (sudormrfbin) [reply helix-term/src/commands.rs:3482]: Oh it's handled in `on_next_key_mode` itself 👍🏾 :

https://github.com/helix-editor/helix/blob/009073fbfa3ef607363ec4e54ee347dcf1ac6398/helix-term/src/commands.rs#L80-L83

**Review by archseer — APPROVED** [review]

**Pull request merged**